pkg/api/validation/validation.go

```diff
@@ -2444,6 +2421,8 @@ func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopo
 	}
 	if len(podAffinityTerm.TopologyKey) != 0 {
 		allErrs = append(allErrs, unversionedvalidation.ValidateLabelName(podAffinityTerm.TopologyKey, fldPath.Child("topologyKey"))...)
+	} else {
```

Comment:
Then I need to remove these tests which you added: https://github.com/kubernetes/kubernetes/pull/47869/files#diff-fb1c11977eba66e7c7c906ab04583f1dR5047